src/utils/validations/component/validations/with-validation.hoc.js

```diff
@@ -0,0 +1,159 @@
+
+const inputWithValidation = (WrappedComponent) => {
+  class WithValidation extends React.Component {
+    static contextType = validationsContext;
+
+    static propTypes = {
+      children: PropTypes.node,
+      name: PropTypes.string,
+      value: PropTypes.string,
+      validate: PropTypes.func,
+      validations: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.arrayOf(PropTypes.func)
+      ]),
+      warnings: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.arrayOf(PropTypes.func)
+      ]),
+      info: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.arrayOf(PropTypes.func)
+      ]),
+      onBlur: PropTypes.func
+    };
+
+    static defaultProps = {
+      validations: [],
+      warnings: [],
+      info: []
+    }
+
+    static contextTypes = {
+      addInput: PropTypes.func,
+      removeInput: PropTypes.func,
+      adjustCount: PropTypes.func
+    }
+
+    state = {
+      hasError: false,
+      hasWarning: false,
+      hasInfo: false
+      // value: ''
+    };
+
+    componentDidMount = () => {
+      if (this.context && this.context.addInput && this.props.validations.length) {
+        this.context.addInput(this.props.name, this.validate);
+      }
+    }
+
+    componentWillUnmount = () => {
+      if (this.context && this.context.removeInput && this.props.validations.length) {
+        this.context.removeInput(this.props.name);
+      }
+    }
+
+    validate = async (types = ['validations', 'warnings', 'info']) => {
+      let result = true;
+      if (types.includes('validations')) result = await this.runValidation('validations');
+      if (result && types.includes('warnings')) result = await this.runValidation('warnings');
+      if (result && types.includes('info')) result = await this.runValidation('info');
+      return result;
+    }
+
+    runValidation = (type) => {
+      return new Promise((resolve) => {
+        if (this.props[type].length === 0) {
+          resolve(true);
+        }
+        const validate = validator(this.props[type]);
+        validate(this.props.value)
+          .then(() => {
+            switch (type) {
+              case 'validations':
+                if (this.state.hasError !== false && this.context.adjustCount) {
+                  this.setState({ hasError: false });
+                  this.context.adjustCount('error', -1);
+                }
+                break;
+              case 'warnings':
+                if (this.state.hasWarning !== false && this.context.adjustCount) {
+                  this.setState({ hasWarning: false });
+                  this.context.adjustCount('warning', -1);
+                }
+                break;
+              case 'info':
+                if (this.state.hasInfo !== false && this.context.adjustCount) {
+                  this.setState({ hasInfo: false });
+                  this.context.adjustCount('info', -1);
+                }
+                break;
+              default:
```

Comment:
that could be removed as it does nothing

Reply:
The linter complains without a default branch, I've added an action so it does something on default now

Reply:
you could try to just add a comment:
`    // no default`
and linter should be happy :)

Reply:
I've added the ignore comments for now open to changing it if anyone else doesn't like it